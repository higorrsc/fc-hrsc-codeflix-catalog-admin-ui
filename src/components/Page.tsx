import { Box, Paper, Typography } from '@mui/material';
import React from 'react';

interface PageProps {
  title: string;
  children: React.ReactNode;
}

export function Page({ title, children }: PageProps) {
  return (
    <Box>
      <Paper>
        <Box p={2}>
          <Box mb={2}>
            <Typography variant='h4'>{title}</Typography>
          </Box>
          {children}
        </Box>
      </Paper>
    </Box>
  );
}
