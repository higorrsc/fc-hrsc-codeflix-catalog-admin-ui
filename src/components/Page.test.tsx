import { render } from '@testing-library/react';
import { Page } from './Page';

describe('Test Page', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Page title='Test'>
        <div>Hello, World!</div>
      </Page>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
