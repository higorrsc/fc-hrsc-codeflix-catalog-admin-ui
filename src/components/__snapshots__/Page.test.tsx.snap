// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Page should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1gtchvp-MuiPaper-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiBox-root css-19midj6"
      >
        <div
          class="MuiBox-root css-1qm1lh"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
          >
            Test
          </h4>
        </div>
        <div>
          Hello, World!
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
